engine/scene/sceneloader.cpp

```diff
@@ -0,0 +1,198 @@
+#include "sceneloader.h"
+
+#include "ball.h"
+#include "button.h"
+#include "checkbox.h"
+#include "colliderbox.h"
+#include "componentdeserializer.h"
+#include "config.h"
+#include "gameobject.h"
+#include "logger.h"
+#include "mesh.h"
+#include "player.h"
+#include "opponent.h"
+#include "scorearea.h"
+#include "scorecontroller.h"
+#include "settingsscreencontroller.h"
+#include "slider.h"
+#include "text.h"
+#include "titlescreencontroller.h"
+#include "transform.h"
+
+#include <nlohmann/json.hpp>
+
+#include <chrono>
+#include <fstream>
+
+namespace pong
+{
+
+SceneLoader::SceneLoader()
+{
+    // Engine components
+    mComponentMappings["Transform"] = [](GameObject* gameObject) {
```

Comment:
Lots of duplication in this function. Is it worth it to abstract these lambdas to some template function? Or do you expect them to be different over time?

Reply:
Just to make sure I know what you mean some thing like 
`mComponentMappings["Transform"] = AddComponentToObject<Transform>(object);`?
I think that would make sense, I don't really expect it to change much.

Reply:
Yep, something like that

Reply:
Cool, works great!